recipes/indexlr.cpp

```diff
@@ -29,55 +29,67 @@ const static size_t INITIAL_OUTPUT_STREAM_SIZE = 100;
 const static size_t QUEUE_SIZE = 64;
 const static size_t MAX_THREADS = 5;
 const static size_t DEFAULT_THREADS = MAX_THREADS;
+const static size_t MAX_QUALITY = 30;
 
 static void
 print_error_msg(const std::string& msg)
 {
   std::cerr << PROGNAME << ' ' << VERSION << ": " << msg << std::endl;
 }
 
+static void
+print_warning_msg(const std::string& msg)
+{
+  std::cerr << "\033[1;33mWarning: \033[0m" << msg << std::endl;
+}
+
 static void
 print_usage()
```

Comment:
We can use [argparse](https://github.com/p-ranav/argparse) to do this for us. Let's merge this PR for now, I'll create an issue and integrate argparse later.